agent-backend/src/crew/build_crew.py

```diff
@@ -168,6 +172,28 @@ def build_tasks(self):
                     context_task_objs.append(context_task)
             
 
+            if task.storeTaskOutput:
+                def callback(output):
+                    
+                    original_file_path = os.path.join(Path(__file__).resolve().parent.parent, 'outputs', task.taskOutputFileName)
+                    
+                    with open(original_file_path, 'w') as f:
+                        f.write(str(output)) 
+                    storage_provider.upload_local_file(task.taskOutputFileName, self.session_id, is_public=True)
+                    mongo_client.insert_model("taskoutputs", {
+                        "session_id": self.session_id,
+                        "task_id": task.id,
+                        "task_output_file_name": task.taskOutputFileName,
+                    })
+                    signed_url = storage_provider.download_file(task.taskOutputFileName, self.session_id, is_public=True)
+
+
+                    self.send_to_sockets(
+                        text=f"File uploaded successfully. Click this link to download your file [{task.taskOutputFileName}]({signed_url})",
+                        event=SocketEvents.MESSAGE,
+                        chunk_id=str(uuid.uuid4())
+                    )
+            
```

Comment:
You shouldn't write the file to disk and then upload from disk. Instead, try and upload to gcs straight from the file in memory, chatGPT suggestion: https://chatgpt.com/share/b2676d48-5edb-4443-b9a1-5f37f5c539cc